pyproject_metadata/__init__.py

```diff
@@ -209,10 +234,34 @@ def __post_init__(self) -> None:
             )
             raise ConfigurationError(msg)
 
+        if self.license_files is not None and isinstance(self.license, dict):
+            msg = 'license-files must not be used when license is not a SPDX string'
```

Comment:
```suggestion
            msg = '"project.license-files" must not be used when "project.license" is not a SPDX license expression'
```
*SPDX exptression* is the term used in the PEP. I would use it here too for consistency, but it may be a bit cryptic, thus SPDX license expression: verbose but hopefully clear.